python/cuml/cuml/cluster/hdbscan/hdbscan.pyx

```diff
@@ -1165,6 +1180,9 @@ def approximate_predict(clusterer, points_to_predict, convert_dtype=True):
     """
     _check_clusterer(clusterer)
 
+    # Reflect the output type from global settings, the clusterer, or the input
+    cuml.internals.set_api_output_type(clusterer._get_output_type(points_to_predict))
```

Comment:
Wouldn't we have to change that back at some point?

Reply:
`set_api_output_type` only sets it within a single decorated call, the return from `api_return_generic` will reset it.